apps/readest-app/src/pages/api/kosync.ts

```diff
@@ -0,0 +1,49 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { corsAllMethods, runMiddleware } from '@/utils/cors';
+import { KoSyncProxyPayload } from '@/types/kosync';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  await runMiddleware(req, res, corsAllMethods);
+
+  const {
+    serverUrl,
+    endpoint,
+    method,
+    headers: clientHeaders,
+    body: clientBody,
+  } = req.body as KoSyncProxyPayload;
+
+  if (req.method !== 'POST') {
+    return res.status(405).json({ error: 'Method Not Allowed' });
+  }
+
+  if (!serverUrl || !endpoint) {
+    return res.status(400).json({ error: 'serverUrl and endpoint are required' });
+  }
+
+  const targetUrl = `${serverUrl.replace(/\/$/, '')}${endpoint}`;
+
+  try {
+    const response = await fetch(targetUrl, {
+      method: method,
+      headers: {
+        ...clientHeaders,
+        'Accept': 'application/vnd.koreader.v1+json',
+        'Content-Type': 'application/json',
+      },
+      body: clientBody ? JSON.stringify(clientBody) : null,
+    });
+
+    const data = await response.text();
+    res.status(response.status);
+    try {
+      res.json(JSON.parse(data));
+    } catch (e) {

```

Comment:
./src/pages/api/kosync.ts
41:14  Error: 'e' is defined but never used.  @typescript-eslint/no-unused-vars